cmake/CompilerConfiguration.cmake

```diff
@@ -1,9 +1,12 @@
 if (CMAKE_Fortran_COMPILER_ID STREQUAL "GNU")
-  set(CMAKE_Fortran_FLAGS          "${CMAKE_Fortran_FLAGS} -ffree-form -std=f2008ts -fimplicit-none -Werror=aliasing -Werror=ampersand -Werror=c-binding-type -Werror=intrinsic-shadow -Werror=intrinsics-std -Werror=line-truncation -Werror=tabs -Werror=target-lifetime -Werror=underflow -Werror=unused-but-set-parameter -Werror=unused-but-set-variable -Werror=unused-variable -Werror=unused-dummy-argument -Werror=conversion -Werror=zerotrip -Werror=uninitialized -Wno-maybe-uninitialized -Werror=unused-parameter")
-  if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 10)
+  set(CMAKE_Fortran_FLAGS          "${CMAKE_Fortran_FLAGS} -ffree-form -std=f2008ts -fimplicit-none -Werror=aliasing -Werror=ampersand -Werror=c-binding-type -Werror=intrinsic-shadow -Werror=intrinsics-std -Werror=line-truncation -Werror=tabs -Werror=target-lifetime -Werror=underflow -Werror=unused-but-set-parameter -Werror=unused-but-set-variable -Werror=unused-variable -Werror=unused-dummy-argument -Werror=conversion -Werror=zerotrip -Wno-maybe-uninitialized -Werror=unused-parameter")
+  if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 10) # comparison against CXX version rather than GFortran version
     set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -Werror=argument-mismatch")  # gcc 10+ has this automatically
    else ()
-    set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fallow-argument-mismatch") # requires for 10+ for the MPI wrap module
+    set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fallow-argument-mismatch") # required for 10+ (MPI wrap)
+  endif ()
+  if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 13) # comparison against CXX version rather than GFortran version
```

Comment:
It seems we have to extend the condition to cover both cases, i.e.:
```
if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 13) # false positive (allocatable array)
  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -Werror=uninitialized")
  else ()
  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -Wno-error=uninitialized")
endif ()
  ```
  
  Then, the warning seems real, due to #694 (likely the new GCC is more pedantic).